### ISSUE TITLE
Unable to create bucket

### ISSUE THREAD

**Issue by pettermoe95** [fsspec/s3fs#813]
I'm using the S3FileSystem and want to create a new bucket on demand.

I'm currently trying this:
```
file_system = S3FileSystem(endpoint_url=config.host, secret=config.secret_key, key=config.access_key)
file_system.mkdir("s3://new_bucket")
```
But it raises:
```
FileNotFoundError: The specified bucket does not exist
```

Based on the source code and documentation I thought this would simply create a new bucket for me. I am connected to data lake storage account in Azure.

How can I create a new bucket when needed? We have an application that uses LakeFS, and for this we need to dynamically create new repositories etc.

**Comment by martindurant** [comment]
You are right that `mkdir(name)` should create a bucket called "name" and we do have tests for this. Perhaps try without the s3:// prefix?

> I am connected to data lake storage account in Azure.

Azure has an s3 endpoint, or is this your own deployment? Sounds like a complex mix of technologies!

> We have an application that uses LakeFS, and for this we need to dynamically create new repositories etc.

I was under the impression that lakeFS would create things for you as needed, so long as it had the right credentials.

**Comment by pettermoe95** [comment]
I tried without the prefix, but still no luck. It might be that you are right about LakeFS and that this is a permissions problem.

I'm not the one that set up the data lake for us, so I will have to investigate this further to understand the problem better.

Thank you for the quick reply. Now I know that I used the S3FileSystem class correctly and can look into other parts of our system to find a solution.
